File: Android/sources/com/android/billingclient/api/UserChoiceDetails.java
package com.android.billingclient.api;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

@zzh
public final class UserChoiceDetails {
    private final String mOriginalJson;
    private final JSONObject mParsedJson;
    private final List<Product> mProducts;

    public UserChoiceDetails(String str) throws JSONException {
        this.mOriginalJson = str;
        JSONObject jSONObject = new JSONObject(str);
        this.mParsedJson = jSONObject;
        this.mProducts = toProductList(jSONObject.optJSONArray("products"));
    }

    private static List<Product> toProductList(JSONArray jSONArray) {
        ArrayList arrayList = new ArrayList();
        if (jSONArray != null) {
            for (int i = 0; i < jSONArray.length(); i++) {
                JSONObject optJSONObject = jSONArray.optJSONObject(i);
                if (optJSONObject != null) {
                    arrayList.add(new Product(optJSONObject));
                }
            }
        }
        return arrayList;
    }

    public String getExternalTransactionToken() {
        return this.mParsedJson.optString("externalTransactionToken");
    }

    public String getOriginalExternalTransactionId() {
        String optString = this.mParsedJson.optString("originalExternalTransactionId");
        if (optString.isEmpty()) {
            return null;
        }
        return optString;
    }

    public List<Product> getProducts() {
        return this.mProducts;
    }

    @zzh
    public static class Product {
        private final String id;
        private final String offerToken;
        private final String type;

        public Product(String str, String str2, String str3) {
            this.id = str;
            this.type = str2;
            this.offerToken = str3;
        }

        public boolean equals(Object obj) {
            if (this == obj) {
                return true;
            }
            if (!(obj instanceof Product)) {
                return false;
            }
            Product product = (Product) obj;
            return this.id.equals(product.getId()) && this.type.equals(product.getType()) && Objects.equals(this.offerToken, product.getOfferToken());
        }

        public String getId() {
            return this.id;
        }

        public String getOfferToken() {
            return this.offerToken;
        }

        public String getType() {
            return this.type;
        }

        public int hashCode() {
            return Objects.hash(new Object[]{this.id, this.type, this.offerToken});
        }

        public String toString() {
            String str = this.id;
            String str2 = this.type;
            return wj6.n(rae.r("{id: ", str, ", type: ", str2, ", offer token: "), this.offerToken, "}");
        }

        private Product(JSONObject jSONObject) {
            this.id = jSONObject.optString("productId");
            this.type = jSONObject.optString("productType");
            String optString = jSONObject.optString("offerToken");
            this.offerToken = true == optString.isEmpty() ? null : optString;
        }
    }
}

File: IOS/v2/com/android/billingclient/api/UserChoiceDetails.swift
package com.iOS.billingclient.api;.swift
import java.util.ArrayList;.swift
import java.util.List;.swift
import java.util.Objects;.swift
import org.json.JSONArray;.swift
import org.json.JSONException;.swift
import org.json.JSONObject;.swift
@zzh.swift
public final class UserChoiceDetails {.swift
private final String mOriginalJson;.swift
private final JSONObject mParsedJson;.swift
private final List<Product> mProducts;.swift
public UserChoiceDetails(String str) throws JSONException {.swift
this.mOriginalJson = str;.swift
JSONObject jSONObject = new JSONObject(str);.swift
this.mParsedJson = jSONObject;.swift
this.mProducts = toProductList(jSONObject.optJSONArray("products"));.swift
}.swift
private static List<Product> toProductList(JSONArray jSONArray) {.swift
ArrayList arrayList = new ArrayList();.swift
if (jSONArray = null) {.swift
for (int i = 0; i < jSONArray.length(); i++) {.swift
JSONObject optJSONObject = jSONArray.optJSONObject(i);.swift
if (optJSONObject = null) {.swift
arrayList.add(new Product(optJSONObject));.swift
}.swift
}.swift
}.swift
return arrayList;.swift
}.swift
public String getExternalTransactionToken() {.swift
return this.mParsedJson.optString("externalTransactionToken");.swift
}.swift
public String getOriginalExternalTransactionId() {.swift
String optString = this.mParsedJson.optString("originalExternalTransactionId");.swift
if (optString.isEmpty()) {.swift
return null;.swift
}.swift
return optString;.swift
}.swift
public List<Product> getProducts() {.swift
return this.mProducts;.swift
}.swift
@zzh.swift
public static class Product {.swift
private final String id;.swift
private final String offerToken;.swift
private final String type;.swift
public Product(String str, String str2, String str3) {.swift
this.id = str;.swift
this.type = str2;.swift
this.offerToken = str3;.swift
}.swift
public boolean equals(Object obj) {.swift
if (this == obj) {.swift
return true;.swift
}.swift
if ((obj instanceof Product)) {.swift
return false;.swift
}.swift
Product product = (Product) obj;.swift
return this.id.equals(product.getId()) && this.type.equals(product.getType()) && Objects.equals(this.offerToken, product.getOfferToken());.swift
}.swift
public String getId() {.swift
return this.id;.swift
}.swift
public String getOfferToken() {.swift
return this.offerToken;.swift
}.swift
public String getType() {.swift
return this.type;.swift
}.swift
public int hashCode() {.swift
return Objects.hash(new Object[]{this.id, this.type, this.offerToken});.swift
}.swift
public String toString() {.swift
String str = this.id;.swift
String str2 = this.type;.swift
return wj6.n(rae.r("{id: ", str, ", type: ", str2, ", offer token: "), this.offerToken, "}");.swift
}.swift
private Product(JSONObject jSONObject) {.swift
this.id = jSONObject.optString("productId");.swift
this.type = jSONObject.optString("productType");.swift
String optString = jSONObject.optString("offerToken");.swift
this.offerToken = true == optString.isEmpty() ? null : optString;.swift
}.swift
}.swift
}.swift
